code/UI/General/TabMenus/Scoreboard/Entry/ScoreboardEntry.cs

```diff
@@ -12,13 +15,42 @@ public class ScoreboardEntry : Panel
 
 	private Image PlayerAvatar { get; init; }
 	private Label PlayerName { get; init; }
+	private Label Tag { get; init; }
 	private Label Karma { get; init; }
 	private Label Score { get; init; }
 	private Label Ping { get; init; }
 
+	private bool Expanded = false;
+	private List<string> _tags = new() { "Friend", "Suspect", "Avoid", "Kill", "Missing" };
+
+	private void SetTag(string tag)
+	{
+		int i = 0;
+		foreach (string cls in Tag.Class)
+		{
+			if (i >= 2)
+				Tag.RemoveClass(cls);
+			i++;
+		}
```

Comment:
This feels like we are somewhat abusing the ordering of the classes (if I'm understanding what this code does). 

I'm not sure of the solution right now, but we should handle this differently. Maybe we could simply delete the UI element and recreate it with the changes?